### PR TITLE
Users crud

### PR DESCRIPTION
**Pull request by Karim9871**

### CODE REVIEW

File: src/services/usersCRUD/UsersCRUD.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	"net/http"
+	"github.com/gorilla/mux"
+	"strconv"
+	"regexp"
+	"log"
+	"github.com/Nastya-Kruglikova/cool_tasks/src/services/common"
+)
+
+type Users struct {
+	Id int
```
Comment (Nastya-Kruglikova) [inline src/services/usersCRUD/UsersCRUD.go]: Id should be ID

File: src/services/usersCRUD/UsersCRUD.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	return nil
+}
+
+func Delete(id int) error {
+	return nil
+}
+
+func GetUsers (w http.ResponseWriter, r *http.Request) {
+	users, err:=SendUsers()
+	if err!=nil {
```
Comment (Nastya-Kruglikova) [inline src/services/usersCRUD/UsersCRUD.go]: pls use gofmt 
if err != nil {
spaces missed

File: src/services/usersCRUD/UsersCRUD.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+		log.Print(err, " ERROR: Can't get users")
+		common.SendError(w, r, 404, "ERROR: Can't get users", err)
+		return
+	}
+	common.RenderJSON(w, r, users)
+}
+
+func GetUserByID (w http.ResponseWriter, r*http.Request) {
+	params:=mux.Vars(r)
+	idUser,readError:=strconv.Atoi(params["id"])
```
Comment (Nastya-Kruglikova) [inline src/services/usersCRUD/UsersCRUD.go]: IDs are UUID in our project

File: src/services/usersCRUD/UsersCRUD.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	users, err:=SendUsers()
+	if err!=nil {
+		log.Print(err, " ERROR: Can't get users")
+		common.SendError(w, r, 404, "ERROR: Can't get users", err)
+		return
+	}
+	common.RenderJSON(w, r, users)
+}
+
+func GetUserByID (w http.ResponseWriter, r*http.Request) {
```
Comment (Nastya-Kruglikova) [inline src/services/usersCRUD/UsersCRUD.go]: Pls use gofmt (or setup it in your IDE)
Should be r *http.Request
space missed

File: src/services/usersCRUD/UsersCRUD.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+	params:=mux.Vars(r)
```
Comment (Nastya-Kruglikova) [inline src/services/usersCRUD/UsersCRUD.go]: Spaces! gofmt!

File: src/services/usersCRUD/UsersCRUD.go [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+	if err!=nil {
+		log.Print(err, "ERROR: Can't find user with such ID")
+		common.SendError(w, r, 404, "ERROR: Can't find user with such ID", err)
+		return
+	}
+	common.RenderJSON(w, r, user)
+}
+
+func AddUser (w http.ResponseWriter, r*http.Request) {
+	parsingError:=r.ParseForm()
```
Comment (Nastya-Kruglikova) [inline src/services/usersCRUD/UsersCRUD.go]: It is ok to use err and not create new names for it each time

File: src/services/usersCRUD/UsersCRUD.go [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+	addingError:=Add(newUser)
+	if addingError!=nil {
+		log.Print(addingError, " ERROR: Can't add this user")
+		common.SendError(w, r, 400, "ERROR: Can't add this user", addingError)
+		return
+	}
+}
+
+func IsValid (user Users) (bool, string) {
+	err:=""
```
Comment (Nastya-Kruglikova) [inline src/services/usersCRUD/UsersCRUD.go]: err should be error always, not string
if you want to use string, rename it as errMessage

File: src/services/usersCRUD/UsersCRUD.go [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+	if err!=nil {
+		log.Print(err, " ERROR: Can't delete this user")
+		common.SendError(w, r, 404, "ERROR: Can't delete this user", err)
+		return
+	}
+}
+
+
+
+//func main () {
```
Comment (Nastya-Kruglikova) [inline src/services/usersCRUD/UsersCRUD.go]: remove commented code

File: src/services/usersCRUD/UsersCRUD.go [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+		return
+	}
+}
+
+func IsValid(user Users) (bool, string) {
+	errMessage := ""
+	var checkPass = regexp.MustCompile(`^[[:graph:]]*$`)
+	var checkName = regexp.MustCompile(`^[A-Z]{1}[a-z]+$`)
+	var checkLogin = regexp.MustCompile(`^[[:graph:]]*$`)
+	var validPass, validName, validLogin bool
```
Comment (mshto) [inline src/services/usersCRUD/UsersCRUD.go]: You can use one variable - isValid, and assign it to true by default

File: src/services/usersCRUD/UsersCRUD.go [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+import (
+	"net/http"
+	"github.com/gorilla/mux"
+	"regexp"
+	"log"
+	"github.com/Nastya-Kruglikova/cool_tasks/src/services/common"
+	"github.com/satori/go.uuid"
+)
+
+type Users struct {
```
Comment (mshto) [inline src/services/usersCRUD/UsersCRUD.go]: Could you please add json representation? Then you can use :

var newUser Users
json.NewDecoder(r.Body).Decode(&newUser)

**Pull request merged**